packages/cli/src/plugin.js

```diff
@@ -0,0 +1,228 @@
+  const namespaces = plugins.map((p) => p.package.split('/')[0])
+  namespaces.forEach((ns) => {
+    if (!ns.includes('@')) {
+      console.warn(
+        chalk.yellow(
+          `⚠️  Plugin "${ns}" is missing a scope/namespace, it will not be loaded.`
+        )
+      )
+    }
+  })
+}
+
+export async function loadPlugins(yargs) {
+  const { plugins, autoInstall } = getConfig().cli
+
+  // TODO: Remove this when we make the full switch to a total plugin based CLI
+  // If we don't have any plugins in the toml file, add all the current @redwoodjs plugins
+  if (plugins.length === 0) {
+    // TODO: Start splitting up CLI into plugins
+    // Nothing to add here at the moment
+  }
+
+  // Validate plugin list from redwood.toml, will print warnings to the console
+  validatePluginList(plugins)
+
+  // Get a list of all unique namespaces, sorted alphabetically but with @redwoodjs first
+  // we ignore any invalid names i.e. names without an '@'
+  const namespaces = Array.from(
+    new Set([
+      ...plugins
+        .filter((p) => p.package.startsWith('@redwoodjs/'))
+        .map((p) => p.package.split('/')[0])
+        .sort((a, b) =>
+          a.package > b.package ? 1 : b.package > a.package ? -1 : 0
+        ),
+      ...plugins
+        .filter((p) => !p.package.startsWith('@redwoodjs/'))
+        .map((p) => p.package.split('/')[0])
+        .sort((a, b) =>
+          a.package > b.package ? 1 : b.package > a.package ? -1 : 0
+        ),
+    ])
+  ).filter((ns) => ns.includes('@'))
+
+  // If the user is running a help command or no command was given
+  // we want to load all plugins for observability in the help output
+  const processArgv = process.argv.slice(2).join(' ')
+  const showingRootHelp =
+    processArgv === '--help' || processArgv === '-h' || processArgv === ''
+
+  // Filter the namespaces based on the command line args to
+  // reduce the number of plugins we need to load
+  const namespacesInUse = namespaces.filter(
+    (ns) => showingRootHelp || processArgv.includes(ns)
+  )
+  if (namespacesInUse.length === 0) {
+    // If no namespace is in use we're using the default @redwoodjs namespace which
+    // is just an empty string ''
+    namespacesInUse.unshift('@redwoodjs')
+  }
+
+  // TODO: We should have some mechanism to fetch the cache from an online or precomputed
+  // source this will allow us to have a cache hit on the first run of a command
+  let pluginCommandCache = {}
+  try {
+    pluginCommandCache = JSON.parse(
+      fs.readFileSync(
+        path.join(getPaths().generated.base, PLUGIN_CACHE_FILENAME)
+      )
+    )
+  } catch (_error) {
+    // No need to log this error, it's just a cache miss
+    // console.log(error)
+  }
+
+  // We filter plugins based on the first word which depends on if a namespace is in use
+  const firstWord = process.argv[2]?.includes('@')
+    ? process.argv[3]
+    : process.argv[2]
+  const showingNamespaceHelp =
```

Comment:
Sorry for all the nits

```suggestion
  const showNamespaceHelp =
```